#! /bin/sh
patch -p1 -f $* < $0
exit $?

Marcus: New glib dropped pipe() macro.

diff -rup claws-mail-3.7.1-cur/src/common/w32lib.h claws-mail-3.7.1/src/common/w32lib.h
--- claws-mail-3.7.1-cur/src/common/w32lib.h	2009-03-06 07:01:00.000000000 +0100
+++ claws-mail-3.7.1/src/common/w32lib.h	2009-04-02 20:54:46.000000000 +0200
@@ -74,6 +74,7 @@
 #include <windows.h>
 #include <io.h>
 #include <stdio.h>
+#include <glib.h>
 
 #ifdef __MINGW32__
 #include <_mingw.h>
@@ -193,6 +194,10 @@ unsigned int sleep( unsigned int seconds
 int link( const char *oldpath, const char *newpath );
 int usleep( unsigned long usec );
 
+#if GLIB_CHECK_VERSION(2, 13, 2)
+#define pipe(phandles)  _pipe (phandles, 4096, _O_BINARY)
+#endif
+
 /*** stdlib ***/
 long int random( void );
 void srandom( unsigned int seed );
diff -rup claws-mail-3.7.1-cur/src/etpan/etpan-thread-manager.c claws-mail-3.7.1/src/etpan/etpan-thread-manager.c
--- claws-mail-3.7.1-cur/src/etpan/etpan-thread-manager.c	2009-04-02 16:10:46.000000000 +0200
+++ claws-mail-3.7.1/src/etpan/etpan-thread-manager.c	2009-04-02 20:55:09.000000000 +0200
@@ -66,5 +66,9 @@ enum {
   TERMINATE_STATE_DONE,
 };
 
+#if GLIB_CHECK_VERSION (2, 13, 2)
+#define pipe(phandles)  _pipe (phandles, 4096, _O_BINARY)
+#endif
+ 
 struct etpan_thread_manager * etpan_thread_manager_new(void)
 {
